djangocms_blog/forms.py

```diff
@@ -69,3 +69,13 @@ def __init__(self, *args, **kwargs):
             # Don't allow app_configs to be added here. The correct way to add an
             # apphook-config is to create an apphook on a cms Page.
             self.fields['app_config'].widget.can_add_related = False
+
+        if 'app_config' in self.data and self.data['app_config'] \
```

Comment:
Just use `config` variable which is instatiated above. If its not empty, the `app_config` is selected and valid and you can avoid the query at line 75